radixdlt/src/main/java/com/radixdlt/consensus/VertexStore.java

```diff
@@ -386,26 +394,37 @@ private boolean addQC(QuorumCertificate qc) {
 		return true;
 	}
 
-	private void insertVertexInternal(Vertex vertex) throws VertexInsertionException {
+	private VertexMetadata insertVertexInternal(Vertex vertex) throws VertexInsertionException {
 		if (!vertices.containsKey(vertex.getParentId())) {
 			throw new MissingParentException(vertex.getParentId());
 		}
 
+		final Vertex vertexToUse;
+		if (vertex.getParentMetadata().isEndOfEpoch()) {
+			// TODO: Check if current proposal is actually an empty client atom
+			vertexToUse = new Vertex(vertex.getEpoch(), vertex.getQC(), vertex.getView(), null);
```

Comment:
Why do we need to force drop the client atom here?

Reply:
Wouldn't it be better to decline to vote on an epoch change proposal with an atom at the BFT level, and have this as failure case at this level?

Reply:
@clstrfsck yes, that would be better but saving that for the next PR.